apps/nextjs/src/app/[locale]/manage/tools/kubernetes/cluster-dashboard/header-card/header-card.tsx

```diff
@@ -0,0 +1,57 @@
+import { Card, Flex, Text, ThemeIcon } from "@mantine/core";
+import { IconBrandGit, IconCloudShare, IconGeometry } from "@tabler/icons-react";
+
+import { useI18n } from "@homarr/translation/client";
+
+import classes from "./header-card.module.css";
+
+type HeaderTypes = "providers" | "version" | "architecture";
+
+interface HeaderCardProps {
+  headerType: HeaderTypes;
+  value: string;
+}
+
+export function HeaderCard(props: HeaderCardProps) {
+  const t = useI18n();
+
+  function getIcon(headerType: HeaderTypes) {
+    if (headerType === "providers") {
+      return <IconCloudShare size={28} stroke={1.5} />;
+    } else if (headerType === "version") {
+      return <IconBrandGit size={28} stroke={1.5} />;
+    }
+    return <IconGeometry size={28} stroke={1.5} />;
+  }
+
+  function getLabel(headerType: HeaderTypes) {
+    if (headerType === "providers") {
+      return t("kubernetes.cluster.providers");
+    } else if (headerType === "version") {
+      return t("kubernetes.cluster.version");
+    }
+    return t("kubernetes.cluster.architecture");
+  }
```

Comment:
Do not create functions inside component functions, since they will be re-created on every render. Extract them to the outside